plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -481,13 +481,15 @@ public void testDate()
         }
     }
 
-    @Test(dataProvider = "sessionZonesDataProvider")
-    public void testTimestampFromTrino(ZoneId sessionZone)
+    @Test(dataProvider = "zone and boolean")
+    public void testTimestampFromTrino(ZoneId sessionZone, boolean useCreateAsSelect)
```

Comment:
I am not convinced this is a good direction. We were debating removal of zone parameter here.
This would allow us to go without any parameters and have simple flow.

Also, it's very common for a test to execute (S)DTT multiple times, with different setup flows (to cover CTAS vs CT + INS possible differences), so it should be as easy as possible. Firing 3 executes in a row is definitely simpler, than defining a data provider method (which, in this case wouldn't be shared), and having an `?:` in the code.

Reply:
I like the other way as well, but it's harder to tell where a failure comes from if CTAS and create-and-insert have different behaviors. But we can debug that easily enough, so I'll put it back.